ci-operator/step-registry/rosa/nodes/wait-ready/rosa-nodes-wait-ready-commands.sh

```diff
@@ -10,24 +10,109 @@ log(){
       oc describe machine "$machine" -n openshift-machine-api
     done
-    # Get details of nodes not in Ready state
-    for node in $(oc get nodes --no-headers -l node-role.kubernetes.io/worker,node-role.kubernetes.io/infra!= --output jsonpath="{.items[?(@.status.conditions[-1].type!='Ready')].metadata.name}"); do
-      oc describe node "$node"
+    echo
+    
+    # Get details of all nodes 
+    listNodeDetails
+}
+
+# Display Machine Config Pool details
+function listMachineConfigPoolDetails() {
+    log "$(date) - List machine config pools"
+    echo "oc get mcp"
+    oc get mcp
+}
+
+# List details of machinesets, machines and nodes depending on Classic Rosa or HCP cluster
+function listDetails() {
+  if [ "$is_hcp_cluster" = "false" ]; then
+    echo "Listing machine pool config, machine and node details"
+    listMachineConfigPoolDetails
+    listMachineAndNodeDetails
+  else
+    echo "Listing node details only"
+    listNodeDetails
+  fi
+}
+
+# Function introducing sleep and wait for worker nodes to become Ready state
+function waitForReady() {
+    echo "Wait for all nodes to be Ready"
+
+    # Sleep for 2 minutes to allow for the nodes to begin restarting after initial scaling
+    sleep 120
+    # Query the node state until all of the nodes are ready
+    FINAL_NODE_STATE="Pass"
+    max_attempts=60 # Max of 60 attempts with 30 sec wait time in between, for total of 30 min
+    for i in $( seq $max_attempts ); do
+        NODE_STATE="$(oc get nodes || echo "ERROR")"
+        if [[ ${NODE_STATE} == *"NotReady"*  || ${NODE_STATE} == *"SchedulingDisabled"* ]]; then
+            FINAL_NODE_STATE="Fail"
+            echo "Not all nodes have finished restarting - waiting for 30 seconds, attempt ${i}"
+            sleep 30
+        elif [[ ${NODE_STATE} == "ERROR" ]]; then
+            retry_api_count=0
+            max_retry_api_attempts=3
+            if [[ "$retry_api_count" -lt "$max_retry_api_attempts" ]]; then
+                NODE_STATE="$(oc get nodes || echo "ERROR")"
+                while [ "$NODE_STATE" == "ERROR" ]; do
+                  retry_api_count++
+                  break
+                done
+            else
+                echo "Failed to connect to API to retrive node status after $max_retry_api_attempts attempts. Response from server:"
+                oc get nodes
+                echo "Exiting test execution."
+                exit 1
+            fi
+        else
+            echo "All nodes are ready to run workloads."
+            FINAL_NODE_STATE="Pass"
+            break
+        fi
+        export FINAL_NODE_STATE
     done
+
+    if [[ ${FINAL_NODE_STATE} == *"Fail"* ]]; then
+        echo "Waited for 30 min for nodes to become Ready. Some or all nodes are NotReady or have SchedulingDisabled. Exiting test case execution!"
+        listDetails
+        exit 1
+    fi
```

Comment:
Meaningless to validate it here for you do 'ready' validation and want to print more logs in the next steps.

Reply:
This step is to list details of machines, machinesets or nodes when all nodes are NOT READY within 30 min. There is no validation here.